nmdc_server/api.py

```diff
@@ -747,6 +754,116 @@ async def update_submission(
+    headers = {"Authorization": f"Bearer {token}", "Content-Type": "text/plain; charset=utf-8"}
+    studyform = submission.metadata_submission["studyForm"]
+    contextform = submission.metadata_submission["contextForm"]
+    multiomicsform = submission.metadata_submission["multiOmicsForm"]
+    pi = studyform["piName"]
+    piorcid = studyform["piOrcid"]
+    datagenerated = "Yes" if contextform["dataGenerated"] else "No"
+    omicsprocessingtypes = ", ".join(multiomicsform["omicsProcessingTypes"])
+    sampletype = ", ".join(submission.metadata_submission["templates"])
+    sampledata = submission.metadata_submission["sampleData"]
+    numsamples = 0
+    for key in sampledata:
+        numsamples = max(numsamples, len(sampledata[key]))
+
+    # some variable data to supply depending on if data has been generated or not
+    data_ids = []
+    if contextform["dataGenerated"]:
+        data_ids = [
+            "NCBI ID: " + multiomicsform["NCBIBioProjectId"],
+            "GOLD ID: " + multiomicsform["GOLDStudyId"],
+            "Alternative IDs/Names: " + ", ".join(multiomicsform["alternativeNames"]),
+        ]
+
+    else:
+        data_ids = [
+            "JGI IDs: " + multiomicsform["JGIStudyId"],
+            "EMSL IDs: " + multiomicsform["studyNumber"],
+            "Alternative IDs/Names: " + ", ".join(multiomicsform["alternativeNames"]),
+        ]
+
+    # assemble the body of the API request
+    body_lis = [
+        f"Submitter: {user.name}, {user.orcid}",
+        f"Submission ID: {submission.id}",
+        f"Has data been generated: {datagenerated}",
+        f"PI name: {pi}",
+        f"PI orcid: {piorcid}",
+        "Status: Submitted -Pending Review",
+        f"Data types: {omicsprocessingtypes}",
+        f"Sample type:{sampletype}",
+        f"Number of samples:{numsamples}",
+        data_ids[0],
+        data_ids[1],
+        data_ids[2],
+        "Note:",
+    ]
+    body_string = " \n ".join(body_lis)
+    payload_dict = {
+        "title": f"NMDC Submission: {submission.id}",
+        "body": body_string,
+        "assignees": ["mslarae13"],
+    }
+
+    payload = json.dumps(payload_dict)
+
+    # make request and log an error or success depending on reply
+    res = requests.post(url=gh_url, data=payload, headers=headers)
+    if res.status_code != 201:
+        logging.error(f"Github issue creation failed with code {res.status_code}")
+        logging.error(res.reason)
+    else:
+        logging.info(f"Github issue creation successful with code {res.status_code}")
+        logging.info(res.reason)
+        # if issue creation is successful we want to put the issue on a project board
+        # if details for that are supplied
+        issue_node_id = res.json()["node_id"]
+        github_issue_to_project(issue_node_id, settings)
+
+    return res
+
+
+def github_issue_to_project(issue_node_id: str, settings):
+    gh_project_token = settings.gh_project_token
```

Comment:
This step requires a different token than `settings.github_authentication_token`?

Reply:
Frustratingly it does seem to required a different token.
Fine grain access tokens don't seem to support the GraphQL api calls needed (found an issue [here](https://github.com/actions/add-to-project/issues/289)) unless they are organization accounts.
And I couldn't find a setting on the classic access token that allowed posting issues. The closest I saw was writing to team conversations but that wouldn't work for this.

I would be happy to be wrong and see that PAT do allow posting issues and then we could just use one token.